README.md

```diff
@@ -1,12 +1,36 @@
-# lambda-template
-Template for creating lambda repositories
-A template for quickly starting a new AWS lambda project.
+# cfn-cr-sc-bucket-policy
 
-## Naming
-Naming conventions:
-* for a vanilla Lambda: `lambda-<context>`
-* for a Cloudformation Transform macro: `cfn-transform-<context>`
-* for a Cloudformation Custom Resource: `cfn-cr-<context>`
+Cloudformation Custom Resource that creates a S3 BucketPolicy. This is intended
+only for use in ServiceCatalog, as it retrieves an ARN that Service Catalog sets,
+`aws:servicecatalog:provisioningPrincipalArn`, to set as a principal on the
+BucketPolicy.
+
+Inventory of source code and supporting files:
+
+- policy_maker - Code for the application's Lambda function.
+- events - Invocation events that you can use to invoke the function.
+- tests - Unit tests for the application code.
+- template.yaml - A template that defines the application's AWS resources.
+
+## Use in a Cloudformation Template
+Create a custom resource in your cloud formation template. Here's an example:
+```yaml
+  SCS3BucketPolicy:
+    Type: Custom::SCS3BucketPolicy
+    Properties:
+      ServiceToken: !ImportValue
+        'Fn::Sub': '${AWS::Region}-cfn-cr-sc-bucket-policy-FunctionArn'
+      BucketName: !Ref S3Bucket
+      ExtraPrincipals: !Ref S3UserARNs
+      RequireEncryption: True
```

Comment:
maybe in your example snippet for `S3Bucket` resource you can set the `BucketEncryption` property  to match this?  I believe synapse only supports AES256 encryption.

Reply:
Again, this is not Synapse specific. I will add some kind of bucket that matches what you say.

Reply:
This is in example added in commit 324d8af